bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.entsoe.internal;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.entsoe.internal.client.Client;
+import org.openhab.binding.entsoe.internal.client.EntsoeTimeSerie;
+import org.openhab.binding.entsoe.internal.client.Request;
+import org.openhab.binding.entsoe.internal.exception.EntsoeConfigurationException;
+import org.openhab.binding.entsoe.internal.exception.EntsoeResponseException;
+import org.openhab.binding.entsoe.internal.exception.EntsoeResponseMapException;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.TimeSeries;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EntsoeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jørgen Melhus - Initial contribution
+ */
+@NonNullByDefault
+public class EntsoeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EntsoeHandler.class);
+
+    private EntsoeConfiguration config;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    private Map<Instant, Optional<EntsoeTimeSerie>> entsoeTimeSeries;
+
+    private final ZoneId cetZoneId = ZoneId.of("CET");
+    private final ZoneId utcZoneId = ZoneId.of("UTC");
+
+    private ZonedDateTime lastDayAheadReceived = ZonedDateTime.of(LocalDateTime.MIN, cetZoneId);
```

Comment:
I think you also need to initialize with this value in `initialize` to trigger initial update after changing configuration, for example?

Reply:
yep, added to initialize